cleanlab/rank.py

```diff
@@ -427,6 +429,133 @@ def get_label_quality_ensemble_scores(
+
+    Returns
+    -------
+    df : pandas DataFrame in which each row corresponds to one example, with columns:
+                quality_annotator_1, quality_annotator_2, ..., quality_annotator_M, annotator_agreement, consensus_label, quality_of_consensus
+            (Here _1, _2, ..., _M suffixes may be replaced column names used to describe annotators in `labels` input)
+    """
+
+    # Compute the overall label health score for each annotator (part of annotator_stats)
+    overall_label_health_score_df = labels_multiannotator.apply(
+        overall_label_health_score, args=[pred_probs], verbose=False
+    )
+
+    # Compute the number of labels labeled/annotated by each annotator (part of annotator_stats)
+    num_labeled = labels_multiannotator.count()
+
+    # Find the worst labeled class for each annotator (part of annotator_stats)
+    worst_class = labels_multiannotator.apply(get_worst_class, args=[pred_probs])
+
+    # Create annotator_stats DataFrame from its columns
+    annotator_stats = pd.DataFrame(
+        {
+            "overall_quality": overall_label_health_score_df,
+            "num_labeled": num_labeled,
+            "worst_class": worst_class,
+        }
+    )
+
+    # Compute the label quality scores for each annotators' labels
+    label_quality_scores_multiannotator = labels_multiannotator.apply(
+        get_label_quality_scores, args=[pred_probs], **kwargs
+    )
+
+    # Prefix column name referring to the annotators' label quality scores
+    label_quality_scores_multiannotator = label_quality_scores_multiannotator.add_prefix(
+        "quality_of_"
+    )
+
+    # Compute the consensus_labels
+    # TODO: deal with ties, conditional based on consensus_method, consensus_method can be a List[str], add dawid-skene
+    consensus_labels = labels_multiannotator.mode(axis=1)
+
+    # Compute the fraction of annotator disagreeing with the consensus labels
+    # i.e. 1 - fraction of annotator agreeing with the consensus labels
+    annotator_disagreement = labels_multiannotator.assign(consensus_label=consensus_labels).apply(
+        lambda s: 1.0 - s[:-1].value_counts(normalize=True)[s["consensus_label"]],
+        axis=1,
+    )
+
+    # Compute the label quality scores of the consensus labels
+    quality_of_consensus = get_label_quality_scores(
+        consensus_labels.to_numpy(), pred_probs, **kwargs
+    )
+
+    # Concatenate additional columns to the label_quality_scores_multiannotator DataFrame
+    (
+        label_quality_scores_multiannotator["consensus_label"],
+        label_quality_scores_multiannotator["annotator_disagreement"],
+        label_quality_scores_multiannotator["quality_of_consensus"],
+    ) = (consensus_labels, annotator_disagreement, quality_of_consensus)
+
+    return (
+        (label_quality_scores_multiannotator, annotator_stats)
+        if return_annotator_stats
+        else label_quality_scores_multiannotator
+    )
+
+
```

Comment:
Remember to add unit test where some annotators never chose a particular class. This is an edge case that may have bugs

Reply:
Remember to add unit test where some examples only have one annotator.

Reply:
Remember to add unit test with ties.